### ISSUE TITLE
Update is not available!

### ISSUE THREAD

**Issue by bernd-wechner** [tendenci/tendenci#928]
The help spells out how I can update my site:

https://www.tendenci.com/help-files/how-do-i-update-my-site/

alas it makes no sense and is incomplete somehow (as measured by complete confusion I guess).  Here what I see:

![image](https://user-images.githubusercontent.com/7296506/100730305-00ae0580-341e-11eb-82a0-838afc07219b.png)

when logged in as a superuser.

And yet I can see that on github here:

![image](https://user-images.githubusercontent.com/7296506/100730387-191e2000-341e-11eb-9208-b44ca835de6e.png)

That help says: "Users with proper permission can update Tendenci to the latest version." but I am a superuser and I can't find any "proper permission" aside from that (this the area where I find that doc broken - as in what are the proper permissions?)

So how can I update it? I guess I'm going to have to pull it down for github and see how far I get ;-). But I'd love to know where the **Update Tendenci** menu item went.

**Comment by bernd-wechner** [comment]
OK, reviewed my install noted (ha, ha, have had a break from this organisations website for a few weeks) but this works fine:

1. Log onto the webserver (`ssh me@websserver`)
2. activate the venv  (`source venvdir/bin/activate`)
3. Upgrade with pip (`pip install --upgrade tendenci`)
4. Restart uwsgi service (assuming you're running as I am under uwsgi: `sudo service uwsgi restart`)

So now I see:

![image](https://user-images.githubusercontent.com/7296506/100733308-19201f00-3422-11eb-812d-918a3989ad0f.png)

it would be nice if the documented menu item existed still. And the lack of it is a bug!

**Comment by jennyq** [comment]
That help file is out dated. The docs for the minor update is here https://tendenci.readthedocs.io/en/latest/upgrade/point-update.html

Although the `update_tendenci` management command is still there and functional with the standard configuration, the "Update Tendenci" button is removed due to the reason that it may not always work with other servers and configurations.

**Comment by bernd-wechner** [comment]
Thanks for clarifying. We should update the help file page then to say this. I'm happy if I can to do that some time, but need to better understand. Is  **Update Tendenci** button permanently gone (so not there for major upgrade either?

What assumptions does the `update_tendenci` management command make? Where can I check the source for that? I'll go looking.

In the mean time based on the point-update doc I simply scripted a bash script after testing it all manually, but it does make firm assumptions ;-). As in I don't run under nginx and gunicorn but under lighttpd and uwsgi. I stripped local specifics and the pr forma is just:

```
#!/bin/bash
#
# Upgrade to latest tendenci release.
# Follows the method outlined at:
#    ttps://tendenci.readthedocs.io/en/latest/upgrade/point-update.html

source <venvdir>/tendenci/bin/activate
sudo service uwsgi stop
pip install --upgrade tendenci
cd <sitedir>
pip install -r requirements/dev.txt --upgrade
pip install -r requirements/prod.txt --upgrade
python manage.py migrate
python manage.py deploy
python manage.py clear_cache
sudo service uwsgi start
```

Worked fine, and I thank you for the pointer as I'd only done the firsts the and of course, if tendenci updates depend on any other package updates or database migrations well ... totally right, this is the process. 

I have an idea: Give most of that is generic it can be made fully generic with:

```
<activate venv>
<stop site>
pip install --upgrade tendenci
cd <site directory>
pip install -r requirements/dev.txt --upgrade
pip install -r requirements/prod.txt --upgrade
python manage.py migrate
python manage.py deploy
python manage.py clear_cache
<start site>
```

which suggests that if the configurable commands:

- `<activate venv>`
- `<stop site>`
- `<start site>`
- `<site directory>`

can be configured then the button could lean on those configs and still be a useful way of upgrading without having to get down an ssh to the webserver. 

They could be configured either on the website in site settings, or perhaps more easily put into a the conf dir when installing or even in settings.py in the conf dir.  <site directory> of course isn't needed as it's known to Django already, but the other three could go into settings and if they are defined the site could display an "Update Tendenci" command again.

It should of course ideally be smart, and only display if an update is available else read "Up to date" or such. 

As ever, I don't mind taking a shot at that if a PR is welcome. It doesn't look like a load of work, but on the flip side, not trivial either and I may to get around to it in any hurry. But it's a real asset to tendenci to have easy update support across the diverse install platforms and seems the installation dependencies are few.

**Comment by jennyq** [comment]
Good thoughts. PRs are always welcome. Here is the management command: https://github.com/tendenci/tendenci/blob/master/tendenci/apps/base/management/commands/update_tendenci.py

Another hurdle is the permission. If you follow the installation guide, Tendenci sites are set up to run as www-data user. If the "Update Tendenci" is initiated from web interface, those operations like start or stop services are not something www-data user allowed to do.

**Comment by bernd-wechner** [comment]
Yes indeed. I know Nextcloud does this so I wonder how they breach the permissions hurdle? In the script currently sudo is only needed to stop and start the uwsgi. pip is fine if using a venv owned by the user running the web server. I suspect Nexcloud doesn't face this as it's PHP and PHP can be updated without restarting a service (I think - as in PHP files are loaded as needed and cached but php probably responds to file timestamps in that process, while uwsgi - and possibly gunicorn - need restarting to reload the python site definition).

I'll have a think, but need to file a few bugs against member imports first ;-)
